src/lib/Microsoft.Health.Fhir.R4.Ingest/Service/R4FhirImportService.cs

```diff
@@ -59,19 +59,26 @@ public override async Task ProcessAsync(ILookupTemplate<IFhirTemplate> config, I
                 var ids = await ResourceIdentityService.ResolveResourceIdentitiesAsync(data).ConfigureAwait(false);
 
                 IEnumerable<IObservationGroup> grps = null;
-                try
-                {
-                    grps = _fhirTemplateProcessor.CreateObservationGroups(config, data);
-                }
-                catch (Exception ex)
+
+                using (ITimed createObservationGroupsMs = _logger.TrackDuration(IomtMetrics.CreateObservationGroupsMs()))
                 {
-                    // user story 93303 to include appropriate message context for this error.
-                    throw new FhirDataMappingException(ex.Message, ex, nameof(FhirDataMappingException));
+                    try
+                    {
+                        grps = _fhirTemplateProcessor.CreateObservationGroups(config, data);
+                    }
+                    catch (Exception ex)
+                    {
+                        // user story 93303 to include appropriate message context for this error.
+                        throw new FhirDataMappingException(ex.Message, ex, nameof(FhirDataMappingException));
+                    }
                 }
 
-                foreach (var grp in grps)
+                using (ITimed saveObservationGroupsMs = _logger.TrackDuration(IomtMetrics.SaveObservationGroupsMs()))

```

Comment:
I understand it is tricky in the current code but if we can figure out how to preserve the partition id so we can include it in the metric that would be a benefit.  It would also be nice if we knew the number of observation groups.